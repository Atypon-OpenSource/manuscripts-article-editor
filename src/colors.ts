export const white = '#fffff'
export const black = '#000'

// greys
export const manuscriptsGrey = '#353535'
export const dustyGrey = '#949494'
export const altoGrey = '#d8d8d8'
export const mercuryGrey = '#e2e2e2'
export const seashellGrey = '#f1f1f1'

// blues
export const jellyBeanBlue = '#2a6f9d'
export const manuscriptsBlue = '#7fb5d5'
export const linkWaterBlue = '#d9e0f3'
export const iceBlue = '#d8edf8'
export const powderBlue = '#e0eef9'
export const aliceBlue = '#f1f8ff'

// reds
export const punchRed = '#dc5030'
export const sungloRed = '#e26969'
export const eunryRed = '#d2a1a1'
export const mandysRed = '#f5c1b7'
export const chablisRed = '#fff1f0'

// oranges
export const zestOrange = '#e28327'
export const wheatYellow = '#f7d7b2'
export const butteryYellow = '#fffceb'

// yellows
export const brightSunYellow = '#fdcd47'
export const huskYellow = '#b6a651'
export const salomieYellow = '#ffe08b'

// greens
export const killarneyGreen = '#3a773a'
export const deYorkGreen = '#80be86'
export const pineGladeGreen = '#a8cd95'
export const springGreen = '#b2c0ac'
export const peppermintGreen = '#dff0d7'
