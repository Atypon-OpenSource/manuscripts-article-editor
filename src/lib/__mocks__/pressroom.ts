export const convert = async (form: FormData) => form.get('file') // echo the input
