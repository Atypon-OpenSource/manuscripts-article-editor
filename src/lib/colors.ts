/*!
 * The contents of this file are subject to the Common Public Attribution License Version 1.0 (the “License”); you may not use this file except in compliance with the License. You may obtain a copy of the License at https://mpapp-public.gitlab.io/manuscripts-frontend/LICENSE. The License is based on the Mozilla Public License Version 1.1 but Sections 14 and 15 have been added to cover use of software over a computer network and provide for limited attribution for the Original Developer. In addition, Exhibit A has been modified to be consistent with Exhibit B.
 *
 * Software distributed under the License is distributed on an “AS IS” basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language governing rights and limitations under the License.
 *
 * The Original Code is manuscripts-frontend.
 *
 * The Original Developer is the Initial Developer. The Initial Developer of the Original Code is Atypon Systems LLC.
 *
 * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
 */

import { hasObjectType } from '@manuscripts/manuscript-transform'
import {
  Color,
  ColorScheme,
  Model,
  ObjectTypes,
} from '@manuscripts/manuscripts-json-schema'

export const DEFAULT_COLOR_SCHEME = 'MPColorScheme:default'

export const isColor = hasObjectType<Color>(ObjectTypes.Color)
export const isColorScheme = hasObjectType<ColorScheme>(ObjectTypes.ColorScheme)

export const nextColorPriority = (colors: Color[]) =>
  Math.max(...colors.map((color) => color.priority || 0)) + 10

const getByPrototype = <T extends Model>(
  modelMap: Map<string, Model>,
  prototype: string
) => {
  for (const model of modelMap.values()) {
    if (model.prototype === prototype) {
      return model as T
    }
  }
}

export const buildColors = (
  modelMap: Map<string, Model>,
  colorSchemeID: string = DEFAULT_COLOR_SCHEME
) => {
  // const colorScheme = modelMap.get(colorSchemeID) as ColorScheme | undefined
  const colorScheme = getByPrototype<ColorScheme>(modelMap, colorSchemeID)

  const colors: Color[] = []

  if (colorScheme && colorScheme.colors) {
    for (const colorID of colorScheme.colors) {
      if (colorID) {
        // const color = modelMap.get(colorID) as Color | undefined
        const color = getByPrototype<Color>(modelMap, colorID)

        if (color && color.value) {
          colors.push(color)
        }
      }
    }

    colors.sort((a, b) => (a.priority || 0) - (b.priority || 0))
  }

  return { colors, colorScheme }
}
