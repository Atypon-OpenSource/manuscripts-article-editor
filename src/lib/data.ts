/*!
 * Â© 2019 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { UserProfileWithAvatar } from '@manuscripts/manuscript-editor'
import { UserProfile } from '@manuscripts/manuscripts-json-schema'
import { RxDocument } from 'rxdb'

export const buildUser = async (
  doc: RxDocument<UserProfile>
): Promise<UserProfileWithAvatar> => {
  const item = doc.toJSON() as UserProfileWithAvatar

  const attachment = await doc.getAttachment('image')

  if (attachment) {
    item.avatar = window.URL.createObjectURL(await attachment.getData())
  }

  return item
}

export const buildUserMap = async (docs: Array<RxDocument<UserProfile>>) => {
  const output: Map<string, UserProfileWithAvatar> = new Map()

  for (const doc of docs) {
    const item = await buildUser(doc)

    output.set(item.userID, item)
  }

  return output
}
