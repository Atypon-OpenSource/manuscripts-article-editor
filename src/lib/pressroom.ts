import axios from 'axios'
import config from '../config'
import token from './token'

const client = axios.create({
  baseURL: config.pressroom.url,
})

client.interceptors.request.use(requestConfig => {
  const tokenData = token.get()

  if (config.pressroom.key) {
    requestConfig.headers['Pressroom-API-Key'] = config.pressroom.key
  } else if (tokenData && tokenData.access_token) {
    requestConfig.headers.Authorization = 'Bearer ' + tokenData.access_token
  }

  return requestConfig
})

export const convert = async (
  form: FormData,
  format: string
): Promise<Blob> => {
  const responseType: XMLHttpRequestResponseType = 'blob'

  const response = await client.post('/v1/document/compile', form, {
    responseType,
    headers: {
      'Pressroom-Target-File-Extension': format.replace(/^\./, ''),
    },
  })

  switch (response.status) {
    case 200:
      break

    // TODO: handle authentication failure (401), timeout, too large, etc

    default:
      throw new Error('Something went wrong: ' + response.data)
  }

  return response.data
}
