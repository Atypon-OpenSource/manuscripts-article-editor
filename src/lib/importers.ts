import JSZip from 'jszip'
import { extname } from 'path'
import { FIGURE } from '../transformer/object-types'
import {
  Attachments,
  ContainedModel,
  Figure,
  Model,
  ModelAttachment,
} from '../types/models'
import { generateAttachmentFilename } from './exporter'
import { convert } from './pressroom'

type Importer = (file: File) => Promise<Model[]>

interface Importers {
  [key: string]: Importer
}

export interface JsonModel
  extends ContainedModel,
    Attachments,
    ModelAttachment {
  _id: string
  bundled: boolean
  collection: string
  contentType: string
}

export interface ProjectDump {
  version: string
  data: JsonModel[]
}

const modelHasObjectType = <T extends Model>(objectType: string) => (
  model: Model
): model is T => {
  return model.objectType === objectType
}

export const readManuscriptFromBundle = async (
  zip: JSZip
): Promise<ProjectDump> => {
  const json = await zip.file('index.manuscript-json').async('text')

  return JSON.parse(json)
}

const importProjectBundle = async (result: Blob) => {
  const zip = await new JSZip().loadAsync(result)

  const doc = await readManuscriptFromBundle(zip)

  if (doc.version !== '2.0') {
    throw new Error(`Unsupported version: ${doc.version}`)
  }

  // TODO: validate?
  // TODO: ensure default data is added

  const items = doc.data
    .filter(item => !item.bundled)
    .filter(item => item._id)
    .map(item => {
      delete item._rev
      delete item.collection

      return item
    })

  const folder = zip.folder('Data')

  await Promise.all(
    items
      .filter(modelHasObjectType<Figure>(FIGURE))
      .map(async (item: Figure) => {
        const filename = generateAttachmentFilename(item._id)

        try {
          item.attachment = {
            id: filename, // TODO: original name?
            type: item.contentType || 'image/png',
            data: await folder.file(filename).async('blob'),
          }
        } catch (error) {
          // tslint:disable-next-line:no-console
          console.error(`Could not retrieve attachment from zip: ${error}`)
          // continue without attachment
        }
      })
  )

  return items
}

const importConvertedFile: Importer = async (file: File) => {
  const form = new FormData()
  form.append('file', file)

  const result = await convert(form, '.manuproj')

  // download(result, 'manuscript.zip')

  return importProjectBundle(result)
}

const importers: Importers = {
  '.docx': importConvertedFile,
  '.html': importConvertedFile,
  '.md': importConvertedFile,
  '.manuproj': importProjectBundle,
}

export const openFilePicker = (): Promise<File> =>
  new Promise((resolve, reject) => {
    const input = document.createElement('input')
    input.type = 'file'
    input.accept = Object.keys(importers).join(',')
    input.addEventListener('change', () => {
      if (input.files && input.files.length) {
        resolve(input.files[0])
      } else {
        reject(new Error('No file was received'))
      }
    })
    input.click()
  })

export const importFile = async (file: File) => {
  const extension = extname(file.name)

  return importers[extension](file)
}
