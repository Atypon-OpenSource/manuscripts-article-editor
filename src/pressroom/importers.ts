/*!
 * The contents of this file are subject to the Common Public Attribution License Version 1.0 (the “License”); you may not use this file except in compliance with the License. You may obtain a copy of the License at https://mpapp-public.gitlab.io/manuscripts-frontend/LICENSE. The License is based on the Mozilla Public License Version 1.1 but Sections 14 and 15 have been added to cover use of software over a computer network and provide for limited attribution for the Original Developer. In addition, Exhibit A has been modified to be consistent with Exhibit B.
 *
 * Software distributed under the License is distributed on an “AS IS” basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language governing rights and limitations under the License.
 *
 * The Original Code is manuscripts-frontend.
 *
 * The Original Developer is the Initial Developer. The Initial Developer of the Original Code is Atypon Systems LLC.
 *
 * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
 */

import {
  ModelAttachment,
  parseJATSArticle,
  parseSTSStandard,
} from '@manuscripts/manuscript-transform'
import {
  Bundle,
  ContributorRole,
  Model,
  ObjectTypes,
} from '@manuscripts/manuscripts-json-schema'
import JSZip from 'jszip'
import { flatMap } from 'lodash-es'
import { basename, extname } from 'path'
import { FileExtensionError } from '../lib/errors'
import {
  createNewContributorRoles,
  createNewStyles,
  fetchSharedData,
} from '../lib/templates'
import { cleanItem } from './clean-item'
import { removeUnsupportedData } from './exporter'
import { convert } from './pressroom'

export interface JsonModel extends Model, ModelAttachment {
  _id: string
  bundled?: boolean
  collection?: string
  contentType?: string
}

export interface ProjectDump {
  version: string
  data: JsonModel[]
}

export const readProjectDumpFromArchive = async (
  zip: JSZip
): Promise<ProjectDump> => {
  const json = await zip.file('index.manuscript-json').async('text')

  return JSON.parse(json)
}

const attachmentKeys: { [key in ObjectTypes]?: string } = {
  [ObjectTypes.Figure]: 'image',
  [ObjectTypes.Bundle]: 'csl',
}

const defaultAttachmentContentTypes: { [key in ObjectTypes]?: string } = {
  [ObjectTypes.Figure]: 'image/png',
  [ObjectTypes.Bundle]: 'application/vnd.citationstyles.style+xml',
}

const importProjectArchive = async (result: Blob) => {
  const zip = await new JSZip().loadAsync(result)

  const projectDump = await readProjectDumpFromArchive(zip)

  if (projectDump.version !== '2.0') {
    throw new Error(`Unsupported version: ${projectDump.version}`)
  }

  // TODO: validate?
  // TODO: ensure default data is added
  // TODO: add default bundle (which has no parent bundle)
  // TODO: ensure that pageLayout and bundle are set

  const items = projectDump.data
    .filter(item => !item.bundled)
    .filter(item => item.objectType !== 'MPContentSummary')
    .filter(item => item._id)
    .map(item => cleanItem(item))

  // load attachments from the Data folder

  for (const file of Object.values(zip.files)) {
    if (!file.dir && file.name.startsWith('Data/')) {
      const id = basename(file.name).replace('_', ':')

      const model = items.find(model => model._id === id)

      if (model) {
        const objectType = model.objectType as ObjectTypes

        if (objectType in attachmentKeys) {
          model.attachment = {
            id: attachmentKeys[objectType]!,
            type:
              model.contentType || defaultAttachmentContentTypes[objectType]!,
            data: await file.async('blob'),
          }
        }
      }
    }
  }

  return items
}

interface BundledData {
  bundles: Map<string, Bundle>
  contributorRoles: Map<string, ContributorRole>
  styles: Map<string, Model>
}

export const importBundledData = async (): Promise<BundledData> => {
  const bundles = await fetchSharedData<Bundle>('bundles')
  const contributorRoles = await fetchSharedData<ContributorRole>(
    'contributor-roles'
  )
  const styles = await fetchSharedData<Model>('styles')

  return {
    bundles,
    contributorRoles: createNewContributorRoles(contributorRoles),
    styles: createNewStyles(styles),
  }
}

const convertFile = async (file: File): Promise<Blob> => {
  const form = new FormData()
  form.append('file', file)

  return convert(form, '.manuproj')
}

const parseXMLFile = async (file: File): Promise<Document> => {
  const url = window.URL.createObjectURL(file)
  const text = await fetch(url).then(response => response.text())
  return new DOMParser().parseFromString(text, 'application/xml')
}

const convertXMLDocument = (doc: Document): Model[] => {
  // TODO: check doc.doctype.publicId?
  switch (doc.documentElement.nodeName) {
    case 'article':
      return parseJATSArticle(doc)

    case 'standard':
      return parseSTSStandard(doc)

    default:
      throw new Error('Unsupported XML format')
  }
}

interface FileType {
  extension: string
  mimetypes: string[]
  description: string
}

const fileTypes: FileType[] = [
  {
    extension: '.docx',
    mimetypes: [
      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
    ],
    description: 'DOCX (Microsoft Word)',
  },
  {
    extension: '.doc',
    mimetypes: ['application/msword'],
    description: 'DOC (Microsoft Word)',
  },
  {
    extension: '.html',
    mimetypes: ['text/html'],
    description: 'HTML',
  },
  {
    extension: '.md',
    mimetypes: ['text/markdown', 'text/plain'],
    description: 'Markdown',
  },
  {
    extension: '.manuproj',
    mimetypes: ['application/zip'],
    description: 'Manuscripts Project Bundle',
  },
  {
    extension: '.latex',
    mimetypes: [
      'application/x-latex',
      'application/latex',
      'text/x-latex',
      'text/latex',
      'text/plain',
    ],
    description: 'LaTeX',
  },
  {
    extension: '.tex',
    mimetypes: [
      'application/x-tex',
      'application/tex',
      'text/x-tex',
      'text/tex',
      'text/plain',
    ],
    description: 'TeX',
  },
  {
    extension: '.xml',
    mimetypes: ['application/xml', 'text/xml'],
    description: 'JATS/STS XML',
  },
  {
    extension: '.zip',
    mimetypes: ['application/zip'],
    description: 'ZIP (containing Markdown or LaTeX)', // TODO: could also be XML or HTML
  },
]

export const acceptedFileExtensions = () => {
  return fileTypes.map(item => item.extension)
}

export const acceptedFileDescription = () => {
  return fileTypes.map(item => item.description)
}

export const acceptedMimeTypes = () => {
  return flatMap(fileTypes, item => item.mimetypes)
}

export const openFilePicker = (
  acceptedExtensions: string[] = acceptedFileExtensions()
): Promise<File> =>
  new Promise((resolve, reject) => {
    const input = document.createElement('input')
    input.type = 'file'
    input.accept = acceptedExtensions.join(',')

    const handleFocus = () => {
      window.removeEventListener('focus', handleFocus)

      // This event is fired before the input's change event,
      // and before the input's FileList has been populated,
      // so a delay is needed.
      window.setTimeout(() => {
        if (!input.files || !input.files.length) {
          resolve()
        }
      }, 1000)
    }

    // window "focus" event, fired even if the file picker is cancelled.
    window.addEventListener('focus', handleFocus)

    input.addEventListener('change', () => {
      if (input.files && input.files.length) {
        const [file] = input.files

        const extension = extname(file.name)

        if (acceptedExtensions.includes(extension)) {
          resolve(file)
        } else {
          const error = new FileExtensionError(extension, acceptedExtensions)
          reject(error)
        }
      } else {
        resolve()
      }
    })

    input.click()
  })

export const importFile = async (file: File) => {
  const { name, lastModified } = file

  if (name.endsWith('.manuproj')) {
    const zip = await new JSZip().loadAsync(file)
    await removeUnsupportedData(zip)
    const blob = await zip.generateAsync({ type: 'blob' })
    file = new File([blob], name, { lastModified })
  }

  // TODO: look inside .zip files
  if (name.endsWith('.xml')) {
    return parseXMLFile(file).then(convertXMLDocument)
  }

  const result = await convertFile(file)

  return importProjectArchive(result)
}
