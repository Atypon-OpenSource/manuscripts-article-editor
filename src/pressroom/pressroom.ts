/*!
 * The contents of this file are subject to the Common Public Attribution License Version 1.0 (the “License”); you may not use this file except in compliance with the License. You may obtain a copy of the License at https://mpapp-public.gitlab.io/manuscripts-frontend/LICENSE. The License is based on the Mozilla Public License Version 1.1 but Sections 14 and 15 have been added to cover use of software over a computer network and provide for limited attribution for the Original Developer. In addition, Exhibit A has been modified to be consistent with Exhibit B.
 *
 * Software distributed under the License is distributed on an “AS IS” basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language governing rights and limitations under the License.
 *
 * The Original Code is manuscripts-frontend.
 *
 * The Original Developer is the Initial Developer. The Initial Developer of the Original Code is Atypon Systems LLC.
 *
 * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
 */

import axios, { AxiosError, AxiosResponse, ResponseType } from 'axios'

import config from '../config'
import { fetchScopedToken } from '../lib/api'
import { ExportManuscriptFormat, ImportManuscriptFormat } from './exporter'

const client = axios.create({
  baseURL: config.pressroom.url,
})

client.interceptors.request.use(async (requestConfig) => {
  const { data: token } = await fetchScopedToken('pressroom-js')

  requestConfig.headers.Authorization = `Bearer ${token}`

  return requestConfig
})

const validateResponse = (response: AxiosResponse) => {
  switch (response.status) {
    case 200:
    case 202:
    case 204:
      break

    // TODO: handle authentication failure (401), timeout, too large, etc

    default:
      console.log(response.headers)
      console.log(response.data)
      throw new Error('Something went wrong: ' + response.data)
  }
}

export const importData = async (
  form: FormData,
  sourceFormat: ImportManuscriptFormat,
  headers = {}
): Promise<Blob> => {
  const response = await client.post<Blob>(`/import/${sourceFormat}`, form, {
    responseType: 'blob' as ResponseType,
    headers,
  })

  validateResponse(response)

  return response.data
}

export const exportData = async (
  form: FormData,
  targetFormat: ExportManuscriptFormat | 'bibliography' | 'word',
  retries = 0
): Promise<Blob> => {
  try {
    if (retries > 0) {
      form.append('allowMissingElements', 'true')
    }

    const response = await client.post<Blob>(`/export/${targetFormat}`, form, {
      responseType: 'blob' as ResponseType,
    })

    validateResponse(response)

    return response.data
  } catch (e: AxiosError | any) {
    if (retries > 0 || e.status !== 400) {
      throw e
    }
    return exportData(form, targetFormat, 1)
  }
}
