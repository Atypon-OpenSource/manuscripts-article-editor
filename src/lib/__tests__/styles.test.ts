/*!
 * The contents of this file are subject to the Common Public Attribution License Version 1.0 (the “License”); you may not use this file except in compliance with the License. You may obtain a copy of the License at https://mpapp-public.gitlab.io/manuscripts-frontend/LICENSE. The License is based on the Mozilla Public License Version 1.1 but Sections 14 and 15 have been added to cover use of software over a computer network and provide for limited attribution for the Original Developer. In addition, Exhibit A has been modified to be consistent with Exhibit B.
 *
 * Software distributed under the License is distributed on an “AS IS” basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language governing rights and limitations under the License.
 *
 * The Original Code is manuscripts-frontend.
 *
 * The Original Developer is the Initial Developer. The Initial Developer of the Original Code is Atypon Systems LLC.
 *
 * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
 */

import data from '@manuscripts/examples/data/project-dump.json'
import {
  BorderStyle,
  Color,
  FigureStyle,
  Model,
  ParagraphStyle,
  TableStyle,
} from '@manuscripts/manuscripts-json-schema'
import { buildModelMap } from '../../pressroom/__tests__/util'
import { ProjectDump } from '../../pressroom/importers'
import { isColor } from '../colors'
import {
  buildFigureStyles,
  buildParagraphStyles,
  buildTableStyles,
  isBorderStyle,
} from '../styles'

const modelMap = buildModelMap(data as ProjectDump)

const getModel = <T extends Model>(id: string) =>
  modelMap.get(id) as T | undefined

const buildColors = (modelMap: Map<string, Model>) => {
  const items = new Map<string, Color>()

  for (const model of modelMap.values()) {
    if (isColor(model)) {
      items.set(model._id, model)
    }
  }

  return items
}

const colors = buildColors(modelMap)

const buildBorderStyles = (modelMap: Map<string, Model>) => {
  const items = new Map<string, BorderStyle>()

  for (const model of modelMap.values()) {
    if (isBorderStyle(model)) {
      items.set(model._id, model)
    }
  }

  return items
}

const borderStyles = buildBorderStyles(modelMap)

describe('styles', () => {
  test('build paragraph styles', () => {
    const model = getModel<ParagraphStyle>(
      'MPParagraphStyle:74D17A3A-33C0-43EB-9389-51335C698744'
    )

    if (!model) {
      throw new Error('Model not found!')
    }

    const result = buildParagraphStyles(model, colors)

    expect(result).toMatchSnapshot()
  })

  test('build figure styles', () => {
    const model = getModel<FigureStyle>(
      'MPFigureStyle:12916784-C8A2-414E-919D-490172E82B25'
    )

    if (!model) {
      throw new Error('Model not found!')
    }

    const result = buildFigureStyles(model, colors, borderStyles)

    expect(result).toMatchSnapshot('')
  })

  test('build table styles', () => {
    const model = getModel<TableStyle>(
      'MPTableStyle:08C0E93B-848D-491F-8EB2-A8A0B17714BA'
    )

    if (!model) {
      throw new Error('Model not found!')
    }

    const result = buildTableStyles(model, colors, borderStyles)

    expect(result).toMatchSnapshot('')
  })
})
