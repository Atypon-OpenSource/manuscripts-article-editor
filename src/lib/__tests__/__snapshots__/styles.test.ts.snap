// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles build paragraph styles: build-paragraph-styles 1`] = `
"
      [paragraphstyle=\\"MPParagraphStyle:74D17A3A-33C0-43EB-9389-51335C698744\\"] {
        font-size: 12pt;
        font-style: normal;
        font-weight: normal;
        color: #000;
        text-align: left;
        margin-top: 10pt !important;
        margin-bottom: 10pt !important;
        line-height: 2;
        text-indent: 0pt;
        .list  { 
      counter-reset: list-level-2 0;
    }
.list  > li { 
      counter-increment: list-level-2;
    }
 > li > ol > li::before {
        content: \\"\\" counter(list-level-2, none) \\".\\";
      }
 > li > ul {
        list-style-type: disc;
      }
.list .list  { 
      counter-reset: list-level-3 0;
    }
.list .list  > li { 
      counter-increment: list-level-3;
    }
.list  > li > ol > li::before {
        content: \\"\\" counter(list-level-3, none) \\".\\";
      }
.list  > li > ul {
        list-style-type: disc;
      }
.list .list .list  { 
      counter-reset: list-level-4 0;
    }
.list .list .list  > li { 
      counter-increment: list-level-4;
    }
.list .list  > li > ol > li::before {
        content: \\"\\" counter(list-level-4, decimal) \\".\\";
      }
.list .list  > li > ul {
        list-style-type: disc;
      }
.list .list .list .list  { 
      counter-reset: list-level-5 0;
    }
.list .list .list .list  > li { 
      counter-increment: list-level-5;
    }
.list .list .list  > li > ol > li::before {
        content: \\"\\" counter(list-level-5, decimal) \\".\\";
      }
.list .list .list  > li > ul {
        list-style-type: disc;
      }
.list .list .list .list .list  { 
      counter-reset: list-level-6 0;
    }
.list .list .list .list .list  > li { 
      counter-increment: list-level-6;
    }
.list .list .list .list  > li > ol > li::before {
        content: \\"\\" counter(list-level-6, decimal) \\".\\";
      }
.list .list .list .list  > li > ul {
        list-style-type: disc;
      }

        ul, ol {
          margin-left: 30pt;
          padding-left: 0 !important;
        }
      }

      ol[paragraphstyle=\\"MPParagraphStyle:74D17A3A-33C0-43EB-9389-51335C698744\\"] {
        margin-left: 0pt !important;

        counter-reset: list-level-1 0;> li { counter-increment: list-level-1 };> li::before {
      content: \\"\\" counter(list-level-1, none) \\".\\";
    }
      }

      ul[paragraphstyle=\\"MPParagraphStyle:74D17A3A-33C0-43EB-9389-51335C698744\\"] {
        margin-left: 0pt !important;

        counter-reset: list-level-1;> li { counter-increment: list-level-1 };list-style-type: disc
      }
    "
`;
